migrations/20250227_viz_move/migration.py

```diff
@@ -0,0 +1,249 @@
+import os
+import shutil
+import re
+import redis
+import logging
+
+
+def setup_logging():
+    """Configure logging for the migration script."""
+    logging.basicConfig(
+        level=logging.INFO,
+        format="%(asctime)s - %(levelname)s - %(message)s",
+        datefmt="%Y-%m-%d %H:%M:%S",
+    )
+    return logging.getLogger("migration")
+
+
+def create_backup(base_folder, logger):
+    """Create backups of all output folders before migration."""
+    logger.info("Creating backups of output folders...")
+
+    shutil.copytree(base_folder, base_folder + "_backup")
+    
+    logger.info("Created backup of poppunk_output")
+    
+    
+def get_output_folders(base_folder):
+    # Get all valid output folders
+    all_output_folders = [
+        os.path.join(base_folder, item)
+        for item in os.listdir(base_folder)
+        if os.path.isdir(os.path.join(base_folder, item))
+    ]
+
+    # Filter for folders with network folder (i.e old output folders to update)
+    output_folders = [
+        folder
+        for folder in all_output_folders
+        if os.path.exists(os.path.join(folder, "network"))
+    ]
```

Comment:
Could just do these in one loop I guess and save a few lines, though you would have one unpythonically long line... 
```suggestion
    # Get all valid output folders
    output_folders = [
        os.path.join(base_folder, item)
        for item in os.listdir(base_folder)
        if os.path.isdir(os.path.join(base_folder, item)) and os.path.exists(os.path.join(base_folder, item "network"))
    ]